manager.go

```diff
@@ -71,6 +67,9 @@ type JobRequest struct {
 	JobParams map[string]string
 
 	Architecture string
+	BuildCluster string
+
+	sync.Mutex
```

Comment:
You can't put a lock on the job request, we copy that around.  You have to put the lock on the thing holding the the cached clients and prevent simultaneous writes to the map - the jobManager.  You need to hide accesses to the map under an accessor method that gives you a client and holds the lock while it updates the cache.

Reply:
@smarterclayton I've made the suggested updates.  I'd like to confirm if these couple of instances, of direct calls to `m.requests`, are OK as they are or should they be updated to the use the `requestsClient()` as well:
```
	for _, req := range m.requests {
		if req.RequestedAt.Add(m.maxAge * 2).Before(now) {
			klog.Infof("request %q is expired", req.User)
			m.requestsClient().Delete(req.User)
		}
	}


	klog.Infof("Job sync complete, %d jobs and %d requests", len(m.jobs), len(m.requests))
```